builder/builder.go

```diff
@@ -58,7 +60,7 @@ func getAuthData(filename string) (*dockerAuthData, error) {
 }
 
 // Build Docker Image.
-func (imgBuilder ImageBuilder) BuildImage() error {
+func (imgBuilder ImageBuilder) BuildImage(assignmentEnv bool) error {
```

Comment:
What is the job of 'assignmentEnv' variable?

Reply:
I am using the same function to build image for code-runner server and for code-runner with assignment environment. Hence I have used this as a flag while giving tag names for the built image.
But all this was the initial code, which I have not changed yet. This shall eventually change after implementing code for installation scripts.

Reply:
Okay.
I understand now.

A better way to implement this is by layering the image build process as shown below.
```
- code-runner image.
    - base assignment image.
        - image with assignment specific libraries installed.
```

So, the assignment configuration file could look like this.
```yaml
config:
    - from
        image: assignmentexec/code-runner:v1.0
    - dependencies:
        lang: python
        langVersion: 3.7
        lib:
            - name: numpy
               cmd: pip3 install numpy
            - scipy
               cmd: pip3 install scipy
            - pandas
               cmd: pip3 install pandas
```

1. So, the `assignmnentexec/code-runner:v1.0` image forms the base layer.
2. `lang` and `langVersion` should be required fields. The image with the language compiler/interpreter of the specified version becomes the second layer. For this example, let's call this `assignmentexec/python3.7:latest` image.
3. The `lib` field should be optional. If libraries are provided to be installed, then that is performed on create the last image layer `assignmentexec/python3.7-numpy-scipy-pandas:latest` image.

You might want to use the [Command Pattern](https://en.wikipedia.org/wiki/Command_pattern) for this.

Reply:
Yes. I will do this in next PR